core/network/src/main/java/com/gkp/core/network/interceptor/TaskyRefreshTokenInterceptor.kt

```diff
@@ -1,18 +1,23 @@
 package com.gkp.core.network.interceptor
 
+import com.gkp.auth.domain.session.SessionStorage
 import com.gkp.network.BuildConfig
+import kotlinx.coroutines.runBlocking
 import okhttp3.Interceptor
 import okhttp3.Response
 
-class TaskyRefreshTokenInterceptor : Interceptor {
+class TaskyRefreshTokenInterceptor(
+    private val sessionStorage: SessionStorage
+) : Interceptor {
     override fun intercept(chain: Interceptor.Chain): Response {
-        val request = chain.request()
+        val accessToken = runBlocking {
+            sessionStorage.getAuthInfo().accessToken
+        }
+        val requestBuider = chain.request()
             .newBuilder()
-            .addHeader(
-                "x-api-key",
-                BuildConfig.API_KEY
-            ).build()
+            .applyApiKey()
```

Comment:
It's a refresh token interceptor which adds an API key though. I wouldn't expect that from reading the name.. Given that some endpoints require the API key while not requiring a token, could it make sense to have two separate interceptors for that?

Reply:
I did not get this point?  Do you mean I should not add the api key header for refreshing the token?